packages/contracts-bedrock/test/L2/L2StandardBridge.t.sol

```diff
@@ -496,6 +668,53 @@ contract L2StandardBridge_Bridge_Test is Bridge_Initializer {
         l2StandardBridge.finalizeDeposit(address(L1Token), address(L2Token), alice, alice, 100, hex"");
     }
 
+    /// @dev Tests that `finalizeDeposit` succeeds when depositing ETH under custom gas token.
+    function test_finalizeDeposit_depositingETH_customGasToken_reverts() external {
+        vm.mockCall(address(l1Block), abi.encodeWithSignature("gasPayingToken()"), abi.encode(address(1), uint8(2)));
+
+        vm.mockCall(
+            address(l2StandardBridge.messenger()),
+            abi.encodeWithSelector(CrossDomainMessenger.xDomainMessageSender.selector),
+            abi.encode(address(l2StandardBridge.OTHER_BRIDGE()))
+        );
+
+        // Should emit both the bedrock and legacy events
+        vm.expectEmit(true, true, true, true, address(l2StandardBridge));
+        emit DepositFinalized(address(L1Token), address(L2Token), alice, alice, 100, hex"");
+
+        vm.expectEmit(true, true, true, true, address(l2StandardBridge));
```

Comment:
Are you sure we need all of this setup stuff here? I think it could work with just the expect revert